indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -376,10 +381,6 @@ public DataSegment push(File file, DataSegment segment) throws IOException
             );
           }
           metrics.incrementProcessed();
-
-          if (numRows >= myRowFlushBoundary) {
-            plumber.persist(firehose.commit());
```

Comment:
it seems removing this changes the behavior, rowFlushBoundary is not honored anymore .